### DISCUSSION TITLE
Getting error with CANoe ope() "object has no attribute 'Name'"

### DISCUSSION THREAD

**Discussion by lthacke3** [chaitu-ycr/py_canoe#33] [General]
I'm trying to use py_canoe to automate a process at work. When I use the open() function, CANoe does open, however I get an error that prevents the rest of the program from running (logs provided below). I believe some users complained of a similar issue in the comments section of [your video](https://youtu.be/eN2OCfJo928?si=DSE7pHznHGAteJN9). There are no records for troubleshooting in your documentation, so I thought I could bring the discussion here. Do you know what could be going on here?
*Personal and proprietary info will be edited*

### Code Snippets
**my_program**
```
from py_canoe import CANoe

can = CANoe()

# fails on below line in open() funtion
can.open('C:\\GM VCU Tools\\GMVCU_RBS_CANoe_10_MY24_165_2022.24.6.tar\\GMVCU_RBS_CANoe_10_MY24_165_2022.24.6\\GMVCU_RBS\\RBS\\GMVCU_RBS_CANoe_10.cfg', True, True, False) 
```

**def open()**
```
line 158 pythoncom.CoInitialize()
...
# fails at line 181 in __fetch_canoe_bus_data() function
line 181 self.__canoe_info['bus'] = self.__fetch_canoe_bus_data()
```

**def __fetch_canoe_bus_data()**
```
line 118 self.__canoe_objects['ReplayCollection'] = self.__canoe_objects['Bus'].ReplayCollection
...
# program finally fails on this line, seems as though replay_block.Name doesn't exist
line 124 replay_blocks_dict[replay_block.Name] = replay_block
```

### Error Logs
```
 File "C:\Program Files\JetBrains\PyCharm Community Edition 2023.2.3\plugins\python-ce\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "C:\Users\my_usename\Desktop\my_project\my_folder\my_program.py", line 6, in <module>
    can.open('C:\\GM VCU Tools\\GMVCU_RBS_CANoe_10_MY24_165_2022.24.6.tar\\GMVCU_RBS_CANoe_10_MY24_165_2022.24.6\\GMVCU_RBS\\RBS\\GMVCU_RBS_CANoe_10.cfg', True, True, False)
  File "C:\Users\my_username\AppData\Local\Programs\Python\Python312\Lib\site-packages\py_canoe.py", line 181, in open
    self.__canoe_info['bus'] = self.__fetch_canoe_bus_data()
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\my_username\AppData\Local\Programs\Python\Python312\Lib\site-packages\py_canoe.py", line 124, in __fetch_canoe_bus_data
    replay_blocks_dict[replay_block.Name] = replay_block
                       ^^^^^^^^^^^^^^^^^
  File "C:\Users\my_username\AppData\Local\Programs\Python\Python312\Lib\site-packages\win32com\client\__init__.py", line 582, in __getattr__
    raise AttributeError(
AttributeError: '<win32com.gen_py.CANoe 10.0 Type Library.IReplayBlock instance at 0x1738451862160>' object has no attribute 'Name'
```

**Comment by chaitu-ycr** [comment]
hi @lthacke3 

please share below details to check further:
py_canoe version =?
python version =?
canoe version =?

package tested considering these prerequisites --> https://chaitu-ycr.github.io/py_canoe/002_prerequisites/
i hope your setup fulfill prerequisites.
 
my observations from your error screens:
- it seems you are using older version of py_canoe. please upgrade to latest version `pip install py_canoe --upgrade`
- from canoe configuration name i see you are using older canoe version 10 which is less than the recommended version >=11.

**Comment by chaitu-ycr** [comment]
hi @lthacke3 

lot of cleanups happened in the recent month.
this issue also addressed in the latest release.
please switch to latest version of package

https://github.com/chaitu-ycr/py_canoe/releases/tag/v2.1.0
https://pypi.org/project/py_canoe/
